src/test/java/co/cask/hydrator/salesforce/etl/BaseSalesforceBatchSourceETLTest.java

```diff
@@ -118,37 +137,72 @@ public static void setupTestClass() throws Exception {
                       parentArtifact,
                       SalesforceBatchSource.class);
 
-    partnerConnection = new PartnerConnection(Authenticator.createConnectorConfig(
-      new AuthenticatorCredentials(USERNAME, PASSWORD, CLIENT_ID, CLIENT_SECRET, LOGIN_URL)));
+    AuthenticatorCredentials credentials = SalesforceConnectionUtil.getAuthenticatorCredentials(
+      USERNAME, PASSWORD, CLIENT_ID, CLIENT_SECRET, LOGIN_URL);
+
+    partnerConnection = SalesforceConnectionUtil.getPartnerConnection(credentials);
+    metadataConnection = createMetadataConnection();
   }
 
   @After
-  public void clearSObjects() throws ConnectionException {
-    String[] ids = createdObjectsIds
-      .stream()
-      .map(SaveResult::getId)
-      .collect(Collectors.toList())
-      .toArray(new String[createdObjectsIds.size()]);
+  public void cleanUp() throws ConnectionException {
+    clearSObjects();
+    deleteCustomObjects();
+  }
 
-    partnerConnection.delete(ids);
+  /**
+   * Creates custom object with provided list of custom fields.
+   * Custom object full name will be saved in order to be deleted in the end of test run.
+   *
+   * @param baseName object base name (without `__c` suffix)
+   * @param fields list of custom fields
+   * @return object full name
+   */
+  protected String createCustomObject(String baseName, CustomField[] fields) throws ConnectionException {
+    CustomObject customObject = initCustomObject(baseName, fields);
+    String fullName = customObject.getFullName();
+    customObjects.add(fullName);
+
+    com.sforce.soap.metadata.SaveResult[] results = metadataConnection.createMetadata(new Metadata[]{customObject});
+    for (com.sforce.soap.metadata.SaveResult result : results) {
```

Comment:
Yes, since we already have in class `com.sforce.soap.partner.SaveResult`. One is used for metadata connection (when creating custom objects), another for partner connection (when adding sObjects).